statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -620,11 +649,31 @@ def _estimate_var(self, lags, offset=0, trend='c'):
         if offset < 0: # pragma: no cover
             raise ValueError('offset must be >= 0')
 
-        y = self.y[offset:]
-
-        z = util.get_var_endog(y, lags, trend=trend, has_constant='raise')
-        y_sample = y[lags:]
-
+        endog = self.endog[offset:]
+        exog = None if self.exog is None else self.exog[offset:]
+        z = util.get_var_endog(endog, lags, trend=trend,
+                               has_constant='raise')
+        if exog is not None:
+            # todo: currently only deterministic terms supported (exoglags==0)
+            # and since exoglags==0, x will be an array of size 0.
+            x = util.get_var_endog(exog[-self.nobs:], 0, trend="nc",
+                                   has_constant="raise")
+            x_inst = exog[-self.nobs:]
+            x = np.column_stack((x, x_inst))
+            del x_inst  # free memory
+            temp_z = z
+            z = np.empty((x.shape[0], x.shape[1]+z.shape[1]))
+            z[:, :self.k_trend] = temp_z[:, :self.k_trend]
+            z[:, self.k_trend:self.k_trend+x.shape[1]] = x
+            z[:, self.k_trend+x.shape[1]:] = temp_z[:, self.k_trend:]
+            del temp_z, x  # free memory
+        # the following modification of z is necessary to get the same results
+        # as JMulTi for the constant-term-parameter...
+        for i in range(self.k_trend):
+            if (np.diff(z[:, i]) == 1).all():  # modify the trend-column
+                z[:, i] += lags
```

Comment:
The quadratic trend "ctt" is still on my todo-list. Here I tried to make test cases pass and there are no test cases for "ctt" because JMulTi does not provide this kind of trend.